packages/react-native-windows-init/src/Cli.ts

```diff
@@ -337,88 +357,90 @@ function isProjectUsingYarn(cwd: string): boolean {
+            `Warning: Querying npm to find react-native-windows@${
+              argv.version
+            } failed.  Attempting to continue anyway...`,
+          );
+        } else {
+          const rnwLatestVersion = await getLatestRNWVersion();
+          console.error(
+            `
+  No compatible version of ${chalk.green('react-native-windows')} found.
+  The latest supported version is ${chalk.green(
+    'react-native-windows',
+  )}@${chalk.cyan(rnwLatestVersion)}.
+  Please modify your application to use ${chalk.green(
+    'react-native',
+  )}@${chalk.cyan(
+              getMatchingReactNativeSemVerForReactNativeWindowsVersion(
+                rnwLatestVersion,
+              ),
+            )} or another supported version of ${chalk.green(
+              'react-native',
+            )} and try again.
+  `,
+          );
+          process.exit(EXITCODE_NO_AUTO_MATCHING_RNW);
+        }
+      }
+
+      if (!argv.version) {
+        console.log(
+          `Latest matching version of ${chalk.bold(
             'react-native-windows',
-          )}@${chalk.cyan(rnwLatestVersion)}.
-You can either downgrade your version of ${chalk.green(
-            'react-native',
-          )} to ${chalk.cyan(
-            getMatchingReactNativeSemVerForReactNativeWindowsVersion(
-              rnwLatestVersion,
-            ),
-          )}, or continue with a ${chalk.yellow(
-            'pre-release',
-          )} version of ${chalk.bold('react-native-windows')}.
-`,
+          )} for ${chalk.green('react-native')}@${chalk.cyan(
+            getReactNativeVersion(),
+          )} is ${chalk.green('react-native-windows')}@${chalk.cyan(
+            rnwResolvedVersion,
+          )}`,
         );
 
-        const confirm: boolean = (await prompts({
-          type: 'confirm',
-          name: 'confirm',
-          message: `Do you wish to continue with ${chalk.green(
-            'react-native-windows',
-          )}@${chalk.cyan(rnwResolvedVersion)}?`,
-        })).confirm;
+        if (rnwResolvedVersion && semver.prerelease(rnwResolvedVersion)) {
+          const rnwLatestVersion = await getLatestRNWVersion();
+          console.warn(
+            `
+  ${chalk.green('react-native-windows')}@${chalk.cyan(
+              rnwResolvedVersion,
+            )} is a ${chalk.yellow('pre-release')} version.
+  The latest supported version is ${chalk.green(
+    'react-native-windows',
+  )}@${chalk.cyan(rnwLatestVersion)}.
+  You can either downgrade your version of ${chalk.green(
+    'react-native',
+  )} to ${chalk.cyan(
+              getMatchingReactNativeSemVerForReactNativeWindowsVersion(
+                rnwLatestVersion,
+              ),
+            )}, or continue with a ${chalk.yellow(
+              'pre-release',
+            )} version of ${chalk.bold('react-native-windows')}.
+  `,
+          );
 
-        if (!confirm) {
-          process.exit(EXITCODE_USER_CANCEL);
+          const confirm: boolean = (await prompts({
```

Comment:
we don't have any interactive input use cases today, I worry about running into these in CI and getting a hang. Is it necessary to have this confirmation, and if so, can we add a param to confirm that CI can pass?

Reply:
share the worry, shall I file a workitem to add a `--noprompt` or `--interactive false`

---
In reply to: [461837301](https://github.com/microsoft/react-native-windows/pull/5603#discussion_r461837301) [](ancestors = 461837301)

Reply:
Right now we don't hit this because the path isn't taken when a version is explicitly passed, which is what CI does right now. We need to pass an explicit version and bypass version matching logic in the master branch, but it would be nice to not need to for stable branches.

Reply:
Actually, I think we might be able to remove explicit versions all around now. We used to have the issue where master was using RN  0.62 and RNW 0.0, but both are 0.0.0 now.

Reply:
If you do want to add a flag though, yargs knows how to do special patterns for negation. E.g. adding a "--prompt" option defaulted to true would automatically add support for "--noPrompt", "--prompt false", etc.

Reply:
@asklar: Since you brought it up: your call....

Reply:
Filed: #5611

Reply:
thanks!